### PR TITLE
Snowbridge: Synchronize from Snowfork repository (#3761)

### PR DESCRIPTION
**Pull request by claravanstaden**
This is a cherry-pick from master of https://github.com/paritytech/polkadot-sdk/pull/3761, excluding PR https://github.com/Snowfork/polkadot-sdk/pull/118

Expected patches for (1.7.0):
- name: snowbridge-pallet-ethereum-client
- name: snowbridge-pallet-inbound-queue
- name: snowbridge-beacon-primitives
- name: snowbridge-core
- name: snowbridge-runtime-test-common

### CODE REVIEW

**Comment by acatangiu** [comment]
What kind of testing has been done on this backport?

**Comment by claravanstaden** [comment]
- [ ] Local E2E tests (register token + send token passes, finalized headers import)
- [ ] `cargo test`
- [ ] `cargo test all-features`
- [ ] clippy passes
- [ ] benchmarks pass

File: prdoc/pr_3761.prdoc [hunk 1]
```diff
@@ -0,0 +1,25 @@
+title: "Snowbridge: Synchronize from Snowfork repository"
+
+doc:
+  - audience: Runtime Dev
+    description: |
+      This PR improves the beacon client to send the execution header along with the message as proof and removes the verification and storing of all execution headers.
+      If the AH sovereign account is depleted and relayer rewards cannot be paid, the message should still be processed.
```
Comment (acatangiu) [inline prdoc/pr_3761.prdoc:7]: is this still the case?

**Review by acatangiu — APPROVED** [review]

File: prdoc/pr_3761.prdoc (hunk 1 above)
Comment (claravanstaden) [reply prdoc/pr_3761.prdoc]: It is not, removed in https://github.com/paritytech/polkadot-sdk/pull/4829/commits/1877257da7bd8f0e92ad0e8abdbf64ff91865173. :) Good catch

**Comment by acatangiu** [comment]
@Morganamilo the PR description has been updated to contain full set of crates needing patch update publish

**Pull request merged**